client.go

```diff
@@ -100,15 +99,22 @@ type Client struct {
 	eventProcessors []EventProcessor
 	integrations    []Integration
 	Transport       Transport
+	// Logger is an instance of log.Logger that is use to provide debug information about running Sentry Client
+	// can be enabled with `Debug` client option
+	Logger *log.Logger
 }
 
 // NewClient creates and returns an instance of `Client` configured using `ClientOptions`.
 func NewClient(options ClientOptions) (*Client, error) {
+	var logger = log.New(ioutil.Discard, "[Sentry] ", log.LstdFlags)
 	if options.Debug {
 		debugWriter := options.DebugWriter
 		if debugWriter == nil {
 			debugWriter = os.Stdout
 		}
+		// Client Logger
+		logger.SetOutput(debugWriter)
+		// Global Logger (This will get mutated every time a new client is created)
 		Logger.SetOutput(debugWriter)
```

Comment:
Suggestion: move this to `Init`, so that `Client` becomes independent of the global logger.

In `NewClient` / `Client`, we only use a local, unexpected, logger (`Client.logger`).